openadmet/models/eval/binary.py

```diff
@@ -0,0 +1,72 @@
+from sklearn.metrics import precision_recall_curve, auc
+import matplotlib.pyplot as plt
+import numpy as np
+import wandb
+import pandas as pd
+
+from openadmet.models.eval.eval_base import EvalBase, evaluators
+
+
+@evaluators.register("PosthocBinaryMetrics")
+class PosthocBinaryMetrics(EvalBase):
+
+    """
+    Intended to be used for regression-based models to calculate
+    precision and recall metrics for user-input
+
+    Not intended for binary models
+    """
+
+    def evaluate(self, y_true=None, y_pred=None, cutoffs=None, report=False, output_dir=None, **kwargs):
+
+        if y_true is None or y_pred is None:
+            raise ValueError("Must provide y_true and y_pred")
+
+        prs_df, baseline = self.get_precision_recall(y_pred, y_true, cutoffs)
+        self.plot_precision_recall_curve(prs_df, baseline, output_dir)
+        self.plot_aupr(prs_df["AUPR"], cutoffs, output_dir)
+
+        self.report(report, output_dir, prs_df)
+
+    def get_precision_recall(self, y_pred, y_true, cutoffs):
+        prs_df = {'Precision':[], 'Recall':[], 'Cutoff':[], 'AUPR':[]}
+        for c in cutoffs:
+            pred_class = [y > c for y in y_pred]
+            true_class = [y > c for y in y_true]
+            precision, recall, _ = precision_recall_curve(true_class, pred_class)
+            prs_df['Precision'].append(precision)
+            prs_df['Recall'].append(recall)
+            prs_df['Cutoff'].append(c)
+            prs_df['AUPR'].append(auc(precision, recall))
+
+        return(pd.DataFrame(prs_df), np.sum(true_class)/len(true_class))
+
+    def plot_precision_recall_curve(self, prs_df, baseline, output_dir):
+        for cutoff in prs_df["Cutoff"]:
```

Comment:
Probably better to just use `df.iterrows()` than doing the == masking but of course both work.